lib/src/main/java/com/nextcloud/android/sso/api/NextcloudAPI.java

```diff
@@ -141,7 +141,10 @@ public <T> T convertStreamToTargetEntity(InputStream inputStream, Type targetEnt
             } else {
                 result = gson.fromJson(targetReader, targetEntity);
                 if (result == null) {
-                    if (targetEntity == Object.class) {
+                    if(targetEntity == Void.class) {
+                        throw new IllegalStateException("Detected Void return type - please " +
+                                "switch to the new return type \"EmptyResponse\" instead.");
```

Comment:
```suggestion
                        ensureTypeNotVoid(Void.class);
```

Reply:
Suggesting to reuse the same method (looks silly but ensures the same error message)